modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EChain.java

```diff
@@ -233,6 +233,7 @@ private void analyzeWrite(final CompilerSettings settings, final Definition defi
         if (last instanceof ADefLink) {
             final ADefLink lastDef = (ADefLink) last;
             expression.analyze(settings, definition, variables);
+            // TODO: does it make more sense to just re-use last.after instead of using storeValueType?
```

Comment:
Okay cool.  I'll make it an interface for now, and we can convert it to a class again as needed.